libcxx/test/std/re/re.alg/re.alg.match/awk.locale.pass.cpp

```diff
@@ -1,136 +0,0 @@
-//===----------------------------------------------------------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-// <regex>
-
-// template <class BidirectionalIterator, class Allocator, class charT,
-//           class traits>
-//   bool regex_match(BidirectionalIterator first, BidirectionalIterator last,
-//                    match_results<BidirectionalIterator, Allocator>& m,
-//                    const basic_regex<charT, traits>& e,
-//                    regex_constants::match_flag_type flags
-//                                            = regex_constants::match_default);
-
-// TODO: investigation needed
-// TODO(netbsd): incomplete support for locales
-// XFAIL: target={{.*}}-linux-gnu{{.*}}, netbsd, freebsd
-// XFAIL: target={{.*}}-amazon-linux{{.*}}
-// XFAIL: target={{.*}}-apple-{{.*}}
-// REQUIRES: locale.cs_CZ.ISO8859-2
-
-#include <regex>
-#include <cassert>
-#include "test_macros.h"
-#include "test_iterators.h"
-
-#include "platform_support.h" // locale name macros
-
-int main(int, char**)
-{
-    std::locale::global(std::locale(LOCALE_cs_CZ_ISO8859_2));
```

Comment:
It looks to me like we're "losing" quite a bit of coverage here. Would it maybe make sense to have some custom locales for this kind of stuff instead? AFAICT the point of these tests is to make sure the appropriate localization APIs are used, not that a locale behaves in a particular way. That would make our localization tests significantly more robust and avoid a bunch of `#ifdefs` for the behaviour of the different platforms.